vm/devices/storage/disk_nvme/nvme_driver/src/tests.rs

```diff
@@ -309,6 +309,275 @@ async fn test_nvme_save_restore_inner(driver: DefaultDriver) {
+        .await
+        .unwrap();
+
+    let device = NvmeTestInterruptTracker::new(nvme, msi_set, dma_client.clone());
+
+    // Create the NVMe driver
+    let nvme_driver = NvmeDriver::new(&driver_source, CPU_COUNT, device, false)
+        .await
+        .unwrap();
+
+    // Access the io_issuers to force creation of IO queues for different CPUs
+    let io_issuers = nvme_driver.io_issuers();
+
+    // Request IO issuers from different CPUs to demonstrate the stride algorithm
+    let _issuer_0 = io_issuers.get(0).await.unwrap();
+    let _issuer_1 = io_issuers.get(1).await.unwrap();
+    let _issuer_2 = io_issuers.get(2).await.unwrap();
+    let _issuer_3 = io_issuers.get(3).await.unwrap();
+    let _issuer_4 = io_issuers.get(4).await.unwrap();
+    let _issuer_5 = io_issuers.get(5).await.unwrap();
+    let _issuer_6 = io_issuers.get(6).await.unwrap();
+    let _issuer_7 = io_issuers.get(7).await.unwrap();
+
+    // Verify the interrupt distribution
+    // With 8 MSI-X vectors and 32 CPUs, we should see stride-based distribution
+    println!("Interrupt distribution with stride algorithm:");
+    println!("Should see interrupts distributed across CPUs with stride 4 (32/8)");
+    println!("Expected: CPUs 0, 4, 8, 12, 16, 20, 24, 28");
+
+    nvme_driver.shutdown().await;
+}
+
+#[async_test]
+async fn test_nvme_multiple_drivers_coordination(driver: DefaultDriver) {
+    const MSIX_COUNT: u16 = 8;
+    const IO_QUEUE_COUNT: u16 = 64;
+    const CPU_COUNT: u32 = 32;
+
+    let driver_source = VmTaskDriverSource::new(SingleDriverBackend::new(driver));
+
+    // Create two separate NVMe devices with different device IDs
+    let device1 = create_test_device(
+        &driver_source,
+        "device1",
+        MSIX_COUNT,
+        IO_QUEUE_COUNT,
+        CPU_COUNT,
+    )
+    .await;
+    let device2 = create_test_device(
+        &driver_source,
+        "device2",
+        MSIX_COUNT,
+        IO_QUEUE_COUNT,
+        CPU_COUNT,
+    )
+    .await;
+
+    // Create drivers for both devices
+    let nvme_driver1 = NvmeDriver::new(&driver_source, CPU_COUNT, device1, false)
+        .await
+        .unwrap();
+    let nvme_driver2 = NvmeDriver::new(&driver_source, CPU_COUNT, device2, false)
+        .await
+        .unwrap();
+
+    // Force creation of IO queues for both drivers
+    let io_issuers1 = nvme_driver1.io_issuers();
+    let io_issuers2 = nvme_driver2.io_issuers();
+
+    // Request issuers from first 4 CPUs for both drivers
+    for cpu in 0..4 {
+        let _issuer1 = io_issuers1.get(cpu).await.unwrap();
+        let _issuer2 = io_issuers2.get(cpu).await.unwrap();
+    }
+
+    // With the device-specific offset, these two drivers should now distribute
+    // their interrupt vectors to different CPU ranges instead of overlapping
+    println!("Multiple driver coordination test completed");
+    println!("Device 1 and Device 2 should use different CPU offsets due to device ID hashing");
```

Comment:
Show me the output with these settings:

* 96 vCPUs, 8 NVMe devices, each with 11 interrupt vectors
* 10 vCPUs, 8 NVMe devices, each with 10 interrupt vectors
* 4 vCPUs, 1 NVMe device with 4 interrupt vectors